libgdx/src/main/java/org/mini2Dx/libgdx/audio/LibgdxAsyncSoundResult.java

```diff
@@ -0,0 +1,63 @@
+package org.mini2Dx.libgdx.audio;
+
+import com.badlogic.gdx.Gdx;
+import org.mini2Dx.core.Mdx;
+import org.mini2Dx.core.audio.AsyncSoundResult;
+import org.mini2Dx.core.audio.Sound;
+import org.mini2Dx.core.exception.MdxException;
+import org.mini2Dx.core.executor.AsyncResult;
+import org.mini2Dx.core.files.FileHandle;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.lang.reflect.InvocationTargetException;
+import java.util.concurrent.Callable;
+
+public class LibgdxAsyncSoundResult implements AsyncSoundResult {
+
+    FileHandle file;
+    AsyncResult<InputStream> result;
+
+    public LibgdxAsyncSoundResult(FileHandle file){
+        this.file = file;
+        result = Mdx.executor.submit(new Callable<InputStream>() {
+            @Override
+            public InputStream call() throws Exception {
+                return new ByteArrayInputStream(file.readBytes());
+            }
+        });
+    }
+
+    /**
+     * Returns the result object
+     *
+     * @return Null if no result is available
+     */
+    @Override
+    public Sound getResult() {
+        if (!isFinished()){
+            return null;
+        }
+        try {
+            return (Sound) Gdx.audio.getClass().getMethod("newSound", InputStream.class, String.class).invoke(Gdx.audio, result.getResult(), file.path());
+        } catch (IllegalAccessException | InvocationTargetException | NoSuchMethodException e) {
+            Mdx.log.info("WARNING", "Async sound loading not supported on this platform, falling back to loading on main thread...");
+            try {
+                return Mdx.audio.newSound(this.file);
+            } catch (IOException ioException) {
+                throw new MdxException("Error while loading sound", ioException);
+            }
+        }
```

Comment:
Cache the resulting file once it's created in case the developer calls getResult multiple times